src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -837,18 +837,34 @@ QStringList QgsOgrProvider::subLayers() const
   }
   else
   {
+    // In case there is no free opened dataset in the cache, keep the first
+    // layer alive while we iterate over the other layers, so that we can
+    // reuse the same dataset. Can help in a particular with a FileGDB with
+    // the FileGDB driver
+    QgsOgrLayer *firstLayer = nullptr;
```

Comment:
Why not make this a QgsOgrLayerUniquePtr, and move layer to it for the first layer (instead of releasing layer into the raw pointer)

Reply:
Good point. Done